### DISCUSSION TITLE
How should I copy a file from a container to another container?

### DISCUSSION THREAD

**Discussion by csa777** [testcontainers/testcontainers-java#5770] [General]
My use case: I have a containerized service that runs in a container. This service has several integration tests that make use of test containers. Some of these containers need to be able to read a file before starting so in the initialization of the GenericContainer I would use `withCopyFileToContainer(MountableFile.forClassPathResource(...))` to copy a file from the container in which the tests run to the test container, but this does not work. The file is not being copied. I assume `withCopyFileToContainer` only works if it is being invoked on the host.

**Comment by aidando73** [comment]
Hi @csa777 do you have a code snippet of what you mean? I'm having a trouble visualizing your setup.

**Comment by kiview** [comment]
`withCopyFileToContainer` should work for your use case. The JVM process would read the file from the filesystem where the JVM process is running (meaning the process executing the tests) and copy it into the container. I think you need to share some more code so that others understand your issue and can help you.

Maybe you want to first download the file from container A and then copy it into container B?

Note: I converted this issue into a discussion since it is a general question.
